Dnstwist Improvements

edujosemena opened the pull request:
Improved  dnstwist runtime and complexity

Fixes #224 

# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

Modified the DNSTwist script to be to run 2 levels deep on domains and implemented threading on DNStwist call to improve runtime.

## 💭 Motivation and context ##

In previous versions, the scanning was not enough to catch certain cases, this change runs dns twist again on the results of first domain. The run time was also too high.

## 🧪 Testing ##

Passes pytests and pre-commit checks.


## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.
- [x] Internal Code Review

edujosemena committed:
Dnstwist Improvements

Improved  dnstwist runtime and complexity

edujosemena committed:
Improved DNSTWIST

edujosemena committed:
Fixed pre commit issues

edujosemena committed:
new changes

edujosemena committed:
pre-commit issues

edujosemena committed:
Merge branch 'develop' into EM-improve-dnstwist

aloftus23 reviewed (changes requested):
@edujosemena We should move run_dnstwist_new.py into pe_source and name it dnstwist.py. So src/pe_source/dnstwist.py.

Also, all database queries (inserts and selects) should be moved into pe_source/data/pe_db/db_query.py (alot of the functions like connecting to the P&E database are already in there)

edujosemena committed:
Central formatting

Changes location and changed location, also used Central Logging

edujosemena committed:
Merge branch 'EM-improve-dnstwist' of https://github.com/cisagov/pe-reports into EM-improve-dnstwist

edujosemena committed:
Update dnstwist.py

edujosemena committed:
test

test

edujosemena committed:
renamed db_query

renaming db_query to avoid errors

edujosemena committed:
Errors in Setup

edujosemena committed:
error

edujosemena committed:
errors

edujosemena committed:
Errors

Error

edujosemena committed:
lgmt Errors

edujosemena committed:
LGMT Changes

edujosemena committed:
Testing

Added tests

edujosemena committed:
Added dnstwist to requirements

edujosemena committed:
updated dnstwist requirement

edujosemena committed:
added comma

edujosemena committed:
Added Dshield to requirements

edujosemena committed:
Added Common_tlds.dict

edujosemena committed:
pytests

pytests

edujosemena committed:
Add pytests

edujosemena committed:
Merge branch 'develop' into EM-improve-dnstwist

edujosemena committed:
Update test_pe_source.py

edujosemena committed:
Update db_query_source.py

aloftus23 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,271 @@
+
+date = datetime.datetime.now().strftime("%Y-%m-%d")
+
+# cisagov Libraries
```
```suggestion
```

aloftus23 replied:
```suggestion
```

aloftus23 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,271 @@
+                    )
+                    domain_list.append(domain_dict)
+        except Exception:
+            # TODO Issue #265 create custom Exceptions
```
```suggestion
            # TODO: Create custom exceptions.
            # Issue 265: https://github.com/cisagov/pe-reports/issues/265
```

aloftus23 replied:
This is the style we've been following and they want it all consistent.

aloftus23 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,271 @@
+            LOGGER.info("Data inserted using execute_values() successfully..")
+
+        except Exception:
+            # TODO Issue #265 create custom Exceptions
```
```suggestion
            # TODO: Create custom exceptions.
            # Issue 265: https://github.com/cisagov/pe-reports/issues/265
```

aloftus23 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,271 @@
+            attacks = int(str(response).split("attacks: ")[1].split("<")[0])
+            reports = int(str(response).split("reports: ")[1].split("<")[0])
+
+        # check dns-a record in DSheild API
```
```suggestion
        # Check dns-a record in DSheild API
```

aloftus23 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,271 @@
+        dom["dns_aaaa"] = [""]
+    else:
+        LOGGER.info(str(dom["dns_aaaa"][0]))
+        # check IP in Blocklist API
```
```suggestion
        # Check IP in Blocklist API
```

aloftus23 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,271 @@
+            attacks = int(str(response).split("attacks: ")[1].split("<")[0])
+            reports = int(str(response).split("reports: ")[1].split("<")[0])
+
+        # check dns-a record in DSheild API
```
```suggestion
        # Check dns-a record in DSheild API
```

aloftus23 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,271 @@
+        return
+    else:
+        LOGGER.info(str(dom["dns_a"][0]))
+        # check IP in Blocklist API
```
```suggestion
        # Check IP in Blocklist API
```

aloftus23 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,271 @@
+                    sub_domain_uid = getSubdomain(PE_conn, sub_domain)[0]
+                    LOGGER.info(sub_domain_uid)
+                except Exception:
+                    # TODO Issue #265 implement custom Exceptions
```
```suggestion
            # TODO: Create custom exceptions.
            # Issue 265: https://github.com/cisagov/pe-reports/issues/265
```

aloftus23 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,271 @@
+        LOGGER.error("These orgs failed:")
+        LOGGER.error(failures)
+
+    PE_conn.close()
```
```suggestion
    PE_conn.close()

if __name__ == "__main__":
    main()

```

aloftus23 reviewed (commented):
Looks great, just requested a bunch of minor style changes.

edujosemena committed:
Update src/pe_source/dnstwistscript.py

Co-authored-by: aloftus23 <79927030+aloftus23@users.noreply.github.com>

edujosemena committed:
Update src/pe_source/dnstwistscript.py

Co-authored-by: aloftus23 <79927030+aloftus23@users.noreply.github.com>

edujosemena committed:
Update src/pe_source/dnstwistscript.py

Co-authored-by: aloftus23 <79927030+aloftus23@users.noreply.github.com>

edujosemena committed:
Update src/pe_source/dnstwistscript.py

Co-authored-by: aloftus23 <79927030+aloftus23@users.noreply.github.com>

edujosemena committed:
Update src/pe_source/dnstwistscript.py

Co-authored-by: aloftus23 <79927030+aloftus23@users.noreply.github.com>

edujosemena committed:
Update src/pe_source/dnstwistscript.py

Co-authored-by: aloftus23 <79927030+aloftus23@users.noreply.github.com>

edujosemena committed:
Update src/pe_source/dnstwistscript.py

Co-authored-by: aloftus23 <79927030+aloftus23@users.noreply.github.com>

edujosemena committed:
Update src/pe_source/dnstwistscript.py

Co-authored-by: aloftus23 <79927030+aloftus23@users.noreply.github.com>

edujosemena committed:
Update src/pe_source/dnstwistscript.py

Co-authored-by: aloftus23 <79927030+aloftus23@users.noreply.github.com>

aloftus23 committed:
Adding dnstiwst to the module options

cduhn17 committed:
Merge branch 'develop' into EM-improve-dnstwist

cduhn17 commented on src/pe_source/data/common_tlds.dict:
```diff
@@ -0,0 +1,369 @@
+fyi
+buzz
+cam
+tl
```
Is this data something we might consider putting into the database?

edujosemena replied:
The common_tlds.dict is just the default top level domain list that comes with DNSTWIST, i dont think its necessary to put it in the database.

cduhn17 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,269 @@
+    elif str(dom["dns_aaaa"][0]) == "!ServFail":
+        dom["dns_aaaa"] = [""]
+    else:
+        LOGGER.info(str(dom["dns_aaaa"][0]))
```
It may be useful to add some context to the logging statement for clarity in the log file.

cduhn17 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,269 @@
+
+def execute_dnstwist(root_domain, test=0):
+    """Run dnstwist on each root domain."""
+    pathtoDict = str(pathlib.Path(__file__).parent.resolve()) + "/data/common_tlds.dict"
```
If we were to put these common tld into a DB table `pathtoDict` could be replaced with a query and result set.

cduhn17 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,269 @@
+
+
+def run_dnstwist(orgs_list):
+    """Connect to PostgreSQL database."""
```
This docstring appears to be comment for the follow on instruction and not a docstring to describe  the functionality .

cduhn17 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,269 @@
+def run_dnstwist(orgs_list):
+    """Connect to PostgreSQL database."""
+    try:
+        PE_conn = connect()
```
Might consider removing this try/except block since `connect()` already has this built into the function. We should also consider logging the exception from the `connect()` function.

cduhn17 commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,269 @@
+
+    """ Get P&E Orgs """
+    orgs = query_orgs_rev()
+    LOGGER.info(orgs["cyhy_db_name"])
```
Add some context to the value being logged for future use.

edujosemena committed:
Updated Docstrings, And Logging Messages

edujosemena committed:
Precommit Issues

edujosemena committed:
added init

edujosemena committed:
Update __init__.py

edujosemena committed:
pre-commit

edujosemena committed:
Update __init__.py

edujosemena reviewed (commented):
Code review by Andy and Craig

cduhn17 reviewed (approved):
Conversations have been resolved.

edujosemena commented:
@dav3r This PR is ready for review

dav3r commented on src/pe_source/data/common_tlds.dict:
```diff
@@ -0,0 +1,369 @@
+fyi
+buzz
+cam
+tl
```
Can this list be alphabetized and checked for duplicates?  That will at least make it simpler to add/remove items from it in the future.

edujosemena replied:
Alphabetized the list.

edujosemena replied:
commit 086bbc1370dc5fd410bc848655440db45fb35455

dav3r commented on src/pe_source/data/pe_db/db_query_source.py:
```diff
@@ -373,7 +373,6 @@ def query_orgs_rev():
     conn = connect()
     sql = "SELECT * FROM organizations WHERE report_on is True ORDER BY organizations_uid DESC;"
     df = pd.read_sql_query(sql, conn)
-    close(conn)
```
Why is this connection not being closed here anymore?

edujosemena replied:
removed to try and fix bug, found the bug and readding the closing of the connection

edujosemena replied:
commit 086bbc1370dc5fd410bc848655440db45fb35455

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+"""Use DNS twist to fuzz domain names and cross check with a blacklist."""
```
Use consistent language throughout:
```suggestion
"""Use dnstwist to fuzz domain names and cross check with a blocklist."""
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+
+
+def checkBlocklist(dom, sub_domain_uid, source_uid, pe_org_uid, perm_list):
+    """Cross reference the dnstwist results with DShield Blocklist."""
```
Unnecessary capitalization:

```suggestion
    """Cross reference the dnstwist results with DShield blocklist."""
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+    if "original" in dom["fuzzer"]:
+        return None, perm_list
+    elif "dns_a" not in dom:
+        return None, perm_list
```
These conditions can be combined since they return the same thing:
```suggestion
    if "original" in dom["fuzzer"] or "dns_a" not in dom:
        return None, perm_list
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+            return None, perm_list
+        LOGGER.info(str(dom["dns_a"][0]))
+
+        # Check IP in Blocklist API
```
Clarify which blocklist API:
```suggestion
        # Check IP in blocklist.de API
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+            attacks = int(str(response).split("attacks: ")[1].split("<")[0])
+            reports = int(str(response).split("reports: ")[1].split("<")[0])
+
+        # Check IP in DSheild API
```
Fix misspelling:
```suggestion
        # Check IP in DShield API
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+    else:
+        if str(dom["dns_a"][0]) == "!ServFail":
+            return None, perm_list
+        LOGGER.info(str(dom["dns_a"][0]))
```
`str(dom["dns_a"][0])` gets used enough times in this function that I'd recommend saving it to a variable:
```suggestion
        domain_a = str(dom["dns_a"][0])
        if domain_a == "!ServFail":
            return None, perm_list
        LOGGER.info(domain_a)
```

If you accept this suggestion, make sure that you also update all other `str(dom["dns_a"][0])` references below.

dav3r replied:
Resolved in https://github.com/cisagov/pe-reports/commit/51f33ffabc14be9b16f043736f6dc3db654fbc0f.

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+        # Check IP in DSheild API
+        try:
+            results = dshield.ip(str(dom["dns_a"][0]), return_format=dshield.JSON)
+            results = json.loads(results)
```
You might as well consolidate this into one line:
```suggestion
            results = json.loads(dshield.ip(domain_a, return_format=dshield.JSON))
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+            attacks = results["ip"]["attacks"]
+            attacks = int(0 if attacks is None else attacks)
+            malicious = True
+            dshield_attacks = attacks
```
I think this is a bit more Pythonic and cleaner:
```suggestion
            dshield_attacks = int(results["ip"].get("attacks", 0))
            malicious = True
```

I believe this also fixes a bug where the `attacks` variable is being unintentionally overwritten here by the DShield results.  Based on code later, it's my understanding that `attacks` is meant to contain the number of attacks reported via blocklist.de.

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+        # Check IP in Blocklist API
+        response = requests.get(
+            "http://api.blocklist.de/api.php?ip=" + str(dom["dns_aaaa"][0])
+        ).content
```
Similar to above, I'd do:

```suggestion
        domain_aaaa = str(dom["dns_aaaa"][0])
        # Check IP in Blocklist API
        response = requests.get(
            "http://api.blocklist.de/api.php?ip=" + domain_aaaa
        ).content
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+            reports = int(str(response).split("reports: ")[1].split("<")[0])
+        try:
+            results = dshield.ip(str(dom["dns_aaaa"][0]), return_format=dshield.JSON)
+            results = json.loads(results)
```
Similar to above:
```suggestion
            results = json.loads(dshield.ip(damain_aaaa"), return_format=dshield.JSON))
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+            attacks = results["ip"]["attacks"]
+            attacks = int(0 if attacks is None else attacks)
+            malicious = True
+            dshield_attacks = attacks
```
Similar to above:
```suggestion
            dshield_attacks = int(results["ip"].get("attacks", 0))
            malicious = True
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+        "blocklist_attack_count": attacks,
+        "blocklist_report_count": reports,
+        "dshield_record_count": dshield_count,
+        "dshield_attack_count": dshield_attacks,
```
It's always best to alphabetize big blocks of data in a dict:
```suggestion
        "blocklist_attack_count": attacks,
        "blocklist_report_count": reports,
        "data_source_uid": source_uid,
        "date_active": date,
        "domain_permutation": dom["domain"],
        "dshield_attack_count": dshield_attacks,
        "dshield_record_count": dshield_count,
        "fuzzer": dom["fuzzer"],
        "ipv4": domain_a,
        "ipv6": domain_aaaa,
        "mail_server": dom["dns_mx"][0],
        "malicious": malicious,
        "name_server": dom["dns_ns"][0],
        "organizations_uid": pe_org_uid,
        "ssdeep_score": dom["ssdeep_score"],
        "sub_domain_uid": sub_domain_uid,
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+
+def execute_dnstwist(root_domain, test=0):
+    """Run dnstwist on each root domain."""
+    pathtoDict = str(pathlib.Path(__file__).parent.resolve()) + "/data/common_tlds.dict"
```
It's a little nicer-looking to use an f-string here, in my opinion:
```suggestion
    pathtoDict = f"{pathlib.Path(__file__).parent.resolve()}/data/common_tlds.dict"
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+        tld=pathtoDict,
+        format="json",
+        threads=8,
+        domain=root_domain,
```
Another good place to alphabetize:
```suggestion
        domain=root_domain,
        format="json",
        registered=True,
        threads=8,
        tld=pathtoDict,
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+    )
+    if test == 1:
+        return dnstwist_result
+    finalorglist = dnstwist_result + []
```
What is the purpose of this line?
```
finalorglist = dnstwist_result + []
```

edujosemena replied:
so that the for loop in the next line does not iterate over itself infinitely times, Removed the [] as it was unnecessary
.

dav3r replied:
I still see the `+ []` - I think you may have overlooked removing it, but have a look at [this suggestion](https://github.com/cisagov/pe-reports/pull/259/files#r1024159880) - I think it's simpler.

dav3r replied:
Resolved via https://github.com/cisagov/pe-reports/commit/acf2a7a446b4ae09e20473aee8ddfbbcc3afe83a.

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+                threads=8,
+                domain=dom["domain"],
+            )
+            finalorglist += secondlist
```
Remove a single-use variable and alphabetize:
```suggestion
            finalorglist += dnstwist.run(
                domain=dom["domain"],
                format="json",
                registered=True,
                threads=8,
                tld=pathtoDict,
            )
```

dav3r commented on tests/test_pe_source.py:
```diff
@@ -335,5 +336,26 @@ def test_shodan_search(
 
+def test_dnstwistfuzzing():
+    """Test if dnstwist is installed correctly."""
+    assert pe_source.dnstwistscript.execute_dnstwist("a.com", test=1) != []
```
Can we test anything more here than that the output is not an empty list?  This test as written isn't all that reassuring.

edujosemena replied:
commit f13aeb9b862c936b00584c941a5efcaad5df7afc

dav3r replied:
That's an improvement and good enough for me to resolve this conversation, but I'd still challenge you to make the test even more specific if possible.

dav3r commented on tests/test_pe_source.py:
```diff
@@ -335,5 +336,26 @@ def test_shodan_search(
+        "ssdeep_score": "",
+    }
+    test1, test2 = pe_source.dnstwistscript.checkBlocklist(dom, 1, 1, 1, [])
+    assert len(test1) == 16
+    assert len(test2) == 1
```
Can you please explain how the length of these outputs confirms that the blocklist is working correctly?

edujosemena replied:
replaced to check if certain values within the result were correct

edujosemena replied:
commit f13aeb9b862c936b00584c941a5efcaad5df7afc

dav3r replied:
Much better, thanks!

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+    PE_conn = connect()
+    source_uid = getDataSource(PE_conn, "DNSTwist")[0]
+
+    """ Get P&E Orgs """
```
Unnecessary capitalization:
```suggestion
    """ Get P&E orgs """
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+        # Only run on orgs in the org list
+        if pe_org_id in orgs_list or orgs_list == "all":
+
+            LOGGER.info("Running DNSTwist on %s", pe_org_id)
+
+            """Collect DNSTwist data from Crossfeed"""
```
Consistent capitalization:
```suggestion
            LOGGER.info("Running dnstwist on %s", pe_org_id)

            """Collect dnstwist data from Crossfeed"""
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+            except Exception:
+                # TODO: Create custom exceptions.
+                # Issue 265: https://github.com/cisagov/pe-reports/issues/265
+                LOGGER.info("Failed selecting DNSTwist data.", "Warning")
```
Consistent capitalization:
```suggestion
                LOGGER.info("Failed selecting dnstwist data.", "Warning")
```

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,261 @@
+                    values,
+                )
+                PE_conn.commit()
+                LOGGER.info("Data inserted using execute_values() successfully..")
```
Remove extra unnecessary period:
```suggestion
                LOGGER.info("Data inserted using execute_values() successfully.")
```

dav3r reviewed (changes requested):
Please consider my feedback and let me know if you have any questions.

edujosemena committed:
Apply suggestions from code review

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r commented:
@edujosemena - For future reference, we strongly encourage smaller, targeted commits with specific commit messages rather than large commits that resolve multiple items in a single commit like you did with https://github.com/cisagov/pe-reports/pull/259/commits/66b3c334430294fc4e60ed62f82b27f309cf5f08.  In addition to making the commit log more useful, this practice makes reviewing your PR easier for the reviewers.

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,253 @@
+
+        # Check IP in DShield API
+        try:
+            results = json.loads(dshield.ip(domain_a, return_format=dshield.JSON))
```
This line won't work as is unless you accept my [earlier suggestion](https://github.com/cisagov/pe-reports/pull/259/files#r1021779698) to create the `domain_a` variable.

edujosemena replied:
Created the domain_a variable

edujosemena replied:
Latest Commit Solves Issue

edujosemena replied:
commit 51f33ffabc14be9b16f043736f6dc3db654fbc0f

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,253 @@
+    )
+    if test == 1:
+        return dnstwist_result
+    finalorglist = dnstwist_result + []
+    for dom in dnstwist_result:
+        if ("tld-swap" not in dom["fuzzer"]) and ("original" not in dom["fuzzer"]):
+            LOGGER.info("Running again on %s", dom["domain"])
+            finalorglist += dnstwist.run(
+                domain=dom["domain"],
+                format="json",
+                registered=True,
+                threads=8,
+                tld=pathtoDict,
+            )
+    return finalorglist
```
I think this should accomplish what you want to do here without needing a separate `finalorglist` variable, though you should test it to confirm:
```suggestion
    for dom in dnstwist_result.copy():
        if ("tld-swap" not in dom["fuzzer"]) and ("original" not in dom["fuzzer"]):
            LOGGER.info("Running again on %s", dom["domain"])
            dnstwist_result += dnstwist.run(
                domain=dom["domain"],
                format="json",
                registered=True,
                threads=8,
                tld=pathtoDict,
            )
    return dnstwist_result
```

edujosemena replied:
Fixed.

edujosemena replied:
Latest Commit uses dnstwist_result.copy() instead of creating a new variable

edujosemena replied:
commit acf2a7a446b4ae09e20473aee8ddfbbcc3afe83a

edujosemena committed:
Updated to create domain_a variable

edujosemena committed:
Alphabetized common_tlds and readded connection closing

edujosemena committed:
implemented copying interation list for execure dnstwist

dav3r commented:
@edujosemena - Can you please provide the relevant commit(s) in each open conversation so that I can re-review without having to search through all of the recent commits to find the appropriate one?  Doing this as a general rule will make life much easier for reviewers of your code.  Thanks!

edujosemena committed:
Updated fuzzing tests

edujosemena committed:
Fixed test_blocklist to test data

Added a flag for testing blocklist.de because it has been down for the last couple of days

edujosemena committed:
Updated test

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,260 @@
+            return None, perm_list
+        LOGGER.info(domain_a)
+
+        # Check IP in blocklist.de API, if tetsing is true, for testing purposes skip this one as it may go offline
```
Fix typo, remove a redundant clause, and re-word:
```suggestion
        # Check IP in blocklist.de API; if testing is true, skip this one since it occasionally goes offline
```

edujosemena replied:
commit 864d2f186177bfaad4fdd0eb1f97bf1372bf7775

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,260 @@
+            return None, perm_list
+        LOGGER.info(domain_a)
+
+        # Check IP in blocklist.de API, if tetsing is true, for testing purposes skip this one as it may go offline
```
Since you mention that blocklist.de sometimes is unavailable, I'd prefer to see this code handle that situation more gracefully.

edujosemena replied:
commit 864d2f186177bfaad4fdd0eb1f97bf1372bf7775

dav3r replied:
That looks much nicer now. 👍

edujosemena committed:
Implemented Timeout

replaced test flag for api calling with a timeout for testing if the api is online.

dav3r reviewed (approved):
Thanks for addressing my feedback! 🎉

edujosemena committed:
Fix dshield api call

edujosemena committed:
remove dnstwists tests

edujosemena committed:
Revert "remove dnstwists tests"

This reverts commit ccfb2ba741bd0e108098ce4a5e6593b3bce834af.

edujosemena committed:
fix issues with skeleton

edujosemena committed:
Update dnstwist DSHIELD exception

dav3r commented on src/pe_source/dnstwistscript.py:
```diff
@@ -0,0 +1,269 @@
+        # Check IP in DShield API
+        try:
+            results = json.loads(
+                dshield.ip(str(dom["dns_a"][0]), return_format=dshield.JSON)
```
Not quite sure why this change (originally made as part of https://github.com/cisagov/pe-reports/pull/259/commits/66b3c334430294fc4e60ed62f82b27f309cf5f08) was undone in https://github.com/cisagov/pe-reports/pull/259/commits/ab4c34c46e6c97731f805a91b43d3b451861d450:
```suggestion
                dshield.ip(domain_a, return_format=dshield.JSON)
```

dav3r commented on .github/workflows/build.yml:
```diff
@@ -112,7 +112,6 @@ jobs:
       fail-fast: false
       matrix:
         python-version:
-          - "3.6"
```
If you are removing support for Python 3.6, you should also remove [the 3.6 line in setup.py](https://github.com/cisagov/pe-reports/blob/develop/setup.py#L76) and add a comment there explaining why 3.6 cannot be supported.

dav3r replied:
You will also need to update [this line](https://github.com/cisagov/pe-reports/blob/develop/setup.py#L83) as well.

edujosemena replied:
commit efec55caa3fa07b6f9f747208fb131052059d698

dav3r replied:
Since this is a temporary issue until `skeleton-python-library` is updated, you should comment out the lines related to 3.6 (as opposed to removing them) and create an issue in this repo as a reminder to change this back later.  After you create the issue, update the comment to reference it.

edujosemena replied:
created the issue and added it as comment. commit a371a5455ed5aed054e392f1ce0f719c206887c0

dav3r replied:
Thanks @edujosemena - I added a [suggestion](https://github.com/cisagov/pe-reports/pull/259#discussion_r1059432227) to improve the language a bit.

edujosemena committed:
Update src/pe_source/dnstwistscript.py

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

edujosemena committed:
Remove 3.6 support

Support for 3.6 was removed temporarily as tests are not passing with 3.6. Waiting for cisagov/skeleton-python-library to update.

edujosemena committed:
Update dnstwistscript.py

Linting issues

edujosemena committed:
Merge branch 'EM-improve-dnstwist' of https://github.com/cisagov/pe-reports into EM-improve-dnstwist

edujosemena committed:
Update setup.py

edujosemena committed:
Update setup.py

dav3r commented on setup.py:
```diff
@@ -73,14 +73,17 @@ def get_version(version_file):
         "License :: CC0 1.0 Universal (CC0 1.0) Public Domain Dedication",
         # Specify the Python versions you support here. In particular, ensure
         # that you indicate whether you support Python 2, Python 3 or both.
+        # Temporarily Removed support for python 3.6 due to build failures, waiting for merge from python skeleton
+        # see issue Re-implement 3.6 compatibility #374
```
Clean up verbiage and include link to relevant PR:
```suggestion
        # Temporarily remove support for Python 3.6 due to build failures; will be reinstated
        # after https://github.com/cisagov/skeleton-python-library/pull/110 is merged
        # See issue #374 "Re-implement 3.6 compatibility"
```

edujosemena committed:
Update setup.py

Reformat comments

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
